dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/Time.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain
+
+import java.util.concurrent.TimeUnit
+
+internal data class Time(
+    val timestamp: Long = System.currentTimeMillis(),
+    val nanoTime: Long = System.nanoTime()
+)
+
+internal fun Long.asTime(): Time {
+    // Because nanoTime only measures the nanoseconds since the beginning
+    // of the current JVM lifetime, we need to approximate the nanotime we want.
```

Comment:
This is because `System.currentTimeMillis()` measures time since epoch (Jan 1st 1970), whereas `System.nanoTime()` returns ```nanoseconds since some fixed but arbitrary origin time``` (usually since device boot).